system/firewalld.py

```diff
@@ -112,13 +112,197 @@
 # Globals
 #
 fw = None
+msgs = None
 module = None
-fw_offline = False
+changed = False
+immediate = None
+permanent = None
 Rich_Rule = None
+fw_offline = False
+desired_state = None
 FirewallClientZoneSettings = None
```

Comment:
I think this goes a bit overboard with use of globals.  I do think trying a global for modules is a good experiment because we use parts of it (the fail_json() method) globally (in lieu of an exception handler hook really).  Many of these others, though, can be easily passed into functions or be return values from functions as they are only used in main and one other function.

In general, globals make it harder to unittest a piece of code and can obscure the inputs and side effects of the code for other people reading it later.